tests/JsonSchemaMapper.Tests/TestTypes.cs

```diff
@@ -84,14 +93,18 @@ public static IEnumerable<ITestData> GetTestDataCore()
 
         // Enum types
         yield return new TestData<IntEnum>(IntEnum.A, ExpectedJsonSchema: """{"type":"integer"}""");
-        yield return new TestData<StringEnum>(StringEnum.A, ExpectedJsonSchema: """{"type": "string", "enum": ["A","B","C"]}""");
-        yield return new TestData<FlagsStringEnum>(FlagsStringEnum.A, ExpectedJsonSchema: """{"type":"string"}""");
+        // `type` is redundant here since `enum` is exclusive
```

Comment:
We added the `type` keyword to enums following feedback from the Semantic Kernel team that this is required by the Gemini APIs (google's LLM).

Reply:
Then I would say that the Gemini LLM is wrong.  It's not hurting anything; it's redundant and a lot of people come to us asking about it.

There is an argument that `type` in this case can serve as an annotation for code gen scenarios for languages which don't support enums directly.

Reply:
I agree that it's wrong, however supporting function calling in all LLMs that we support is what funds this project. As such, I can't see how we could remove it at this point. I think we could remove it in a future revision though, we should be free to evolve the generated schemas without it being considered a breaking change.